document-store/src/main/java/org/hypertrace/core/documentstore/Datastore.java

```diff
@@ -3,11 +3,16 @@
 import com.typesafe.config.Config;
 import java.util.Map;
 import java.util.Set;
+import org.hypertrace.core.documentstore.model.config.ConnectionConfig;
 
 public interface Datastore {
 
+  /** Deprecated in favour of {@link Datastore#init(ConnectionConfig)} */
+  @Deprecated(forRemoval = true)
   boolean init(Config datastoreConfig);
 
+  void init(final ConnectionConfig connectionConfig);
```

Comment:
Will these change the way we provide config in application.conf files wherever this lib is used? Or its a subset of existing config?

Reply:
I am assuming that main motivation here that to move away from typesafe config, right?

Reply:
Yes. To decouple typesafe config from document-store model

Reply:
For now, we have all the config related to connection - `ConnectionConfig`. and also so far we didn't find a need for other config. Bug, was thinking, would `DocStoreConfig` have been more generic that include `connection` + any other config?

Just a thought, I guess, you would have already consider that, right?

Reply:
That makes sense. Wrapped `ConnectionConfig` inside `DatastoreConfig` for datastore-level configurations. (so that we can later introduce `CollectionConfig` also for collection-level configurations).